mirroroperator/registrymirror.py

```diff
@@ -254,64 +303,53 @@ def generate_daemon_set(self, daemon_set):
 
-    def generate_service(self, service):
-        service.spec.type = "NodePort"
-        return service
-
     def generate_headless_service(self, service_headless):
         service_headless.spec.cluster_ip = 'None'
         service_headless.spec.type = "ClusterIP"
         return service_headless
 
-    def handle_proxy_credentials(self, env):
+    def get_upstream_credentials(self):
         credentials_secret = None
         if self.credentials_secret_name:
             credentials_secret = self.run_action_and_parse_error(self.core_api.read_namespaced_secret,
-                                                             self.credentials_secret_name,
-                                                             self.namespace)
+                                                                 self.credentials_secret_name,
+                                                                 self.namespace)
         if not credentials_secret:
             LOGGER.error("No secret named %s was found in the %s namespace, will use unauth access",
                          self.credentials_secret_name, self.namespace)
-            return env
+            return None
 
         encoded_user = credentials_secret.data.get("username")
         encoded_pass = credentials_secret.data.get("password")
 
         if not (encoded_user and encoded_pass):
             LOGGER.error("Secret %s does not contain username/password",
                          self.credentials_secret_name)
-            return env
-        env.append(client.V1EnvVar(name="REGISTRY_PROXY_USERNAME",
-                                   value=None,
-                                   value_from=client.V1EnvVarSource(
-                                        secret_key_ref=client.V1SecretKeySelector(
-                                            key="username",
-                                            name=self.credentials_secret_name
-                                        )
-                                   ))
-                  )
-        env.append(client.V1EnvVar(name="REGISTRY_PROXY_PASSWORD",
-                                   value=None,
-                                   value_from=client.V1EnvVarSource(
-                                        secret_key_ref=client.V1SecretKeySelector(
-                                            key="password",
-                                            name=self.credentials_secret_name
-                                        )
-                                   ))
-                  )
-        LOGGER.info("Secret selected + env vars set successfully")
-        return env
-
+            return None
+        the_user = base64.b64decode(encoded_user.encode()).decode()
+        the_pass = base64.b64decode(encoded_pass.encode()).decode()
+        LOGGER.info("Secret selected successfully")
+        return (the_user, the_pass)
 
     def generate_stateful_set(self):
-
+        script = '''
+        apk add --no-cache ca-certificates
```

Comment:
At runtime apk should not be being run - there's no guarantee the internet is accessible. (And it's not, in some environments)

Reply:
OK. We need to publish an alpine docker image with those certificates installed then, if one doesn't exist already. If necessary the CI pipeline of this project could be used to do so.

Reply:
https://gitlab.tech.lastmile.com/internal-open-source/alpine-base-images

Reply:
To clarify, these certificates are not needed by the mirror operator - they are needed by the individual registry mirrors.